internal/connect/hwinfo.go

```diff
@@ -269,3 +273,44 @@ func readValues2map(b []byte) map[string]string {
 	}
 	return m
 }
+
+// Returns the total memory in megabytes if the given `line` is from the
+// `MemTotal` key of the `/proc/meminfo` file.
+func parseTotal(line string) (int, error) {
+	fields := strings.Fields(line)
+	if len(fields) < 1 {
+		return 0, errors.New("memory information not found")
+	}
+	if fields[0] != "MemTotal:" {
+		return 0, fmt.Errorf("skipping field '%v'", fields[0])
+	}
+
+	val, err := strconv.Atoi(fields[1])
+	if err != nil {
+		return 0, err
+	}
+	return val / 1024, nil
+}
+
+// Returns an integer with the amount of megabytes of total memory (i.e.
+// `MemTotal` in /proc/meminfo). It will return 0 if this information could not
+// be extracted for whatever reason.
+func systemMemory() int {
+	file, err := os.Open("/proc/meminfo")
+	if err != nil {
+		Debug.Print("'/proc/meminfo' could not be read!")
+		return 0
+	}
+	defer file.Close()
```

Comment:
You might consider passing `file` to a function that takes an `io.Reader` and that function will do everything else. Then there can be unit tests too. See https://github.com/SUSE/connect-ng/blob/b225bc392fef154bc2e5a69007f6695901e596fe/internal/connect/config.go#L84 and `config_test.go`